docs/pii-encryption.md

```diff
@@ -0,0 +1,126 @@
+# login.gov PII Cryptography Overview
+
+`login.gov` is a service that provides single-sign-on capability to
+users of US government websites. It is developed jointly by 18F and USDS.
+Its goals include:
+
+* simplifying online interactions with the US government, by
+  reducing the number of usernames and passwords people need to remember;
+* improving security, by reducing the number of passwords people need to memorize
+  and implementing multi-factor authentication;
+* helping government online services avoid the need to develop and host their
+  own authentication systems, by serving as a federated identity-as-a-service.
+
+Because the long term vision of `login.gov` is ubiquitous use across
+government, it is expected that someday nearly all Americans will have a
+`login.gov` account. This means that securely storing sensitive personally identifiable
+information (PII), such as Social Security Numbers (SSN) or passwords, is critical.
+
+This document outlines the technical measures `login.gov` has implemented to
+meet its responsibility for the secure storage of this information.
+
+## Design considerations
+
+Almost everyone will agree that sensitive information should be encrypted.
+However, merely saying "data is encrypted" is insufficient in real-world
+systems; instead we have to be concerned with implementation specifics,
+including key management.
+
+Many systems encrypt data at rest with a single online key.
+Unfortunately, many attacks which compromise the
+ciphertext will also compromise the key, so these designs provide limited
+protection.
+
+We delineate two kinds of PII: sensitive and non-sensitive. Non-sensitive PII
+consists of email address and phone number, both used for multi-factor authentication.
+Sensitve PII includes everything else a verified user might register with us:
+full name, address, date of birth, and SSN.
+
+Therefore, we have adopted the following design goals:
+
+* Encryption should rely on a multi-factor model, just like authentication does.
+* Long term sensitive PII storage should be encrypted under a key derived from the user's
+  password.
+* Short term (session length) sensitive PII storage may be encrypted
+  under a key controlled exclusively by the server.
+* Sensitive PII only needs to be available during the user session. Offline/background
+  access is not necessary. The exception is non-sensitive PII (email and phone).
+* Forgetting your password should not necessarily result in all PII being lost.
```

Comment:
should we also include the goal of disallowing duplicate accounts and/or fraud detection via SSN?